src/java/org/apache/cassandra/net/Verb.java

```diff
@@ -630,6 +634,16 @@ private static int idForCustomVerb(int id)
     {
         return CUSTOM_VERB_START - id;
     }
+
+    private static <A, B> IVersionedAsymmetricSerializer<A, B> accordEmbedded(AsymmetricVersionedSerializer<A, B, Version> delegate)
```

Comment:
something now properly handled in this patch is figuring out what version to use.  In some code paths we have relied on node version to infer what capabilities are present.  There is a ML thread about adding capabilities into C*, which would let us denote what versions are supported by which notes...  for now this logic is mostly a place holder